packages/x-data-grid-premium/src/components/GridPinnedRows.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import composeClasses from '@mui/utils/composeClasses';
+import { getDataGridUtilityClass, gridClasses, useGridSelector } from '@mui/x-data-grid-pro';
+import {
+  GridPinnedRowsProps,
+  gridPinnedRowsSelector,
+  gridRenderContextSelector,
+  useGridPrivateApiContext,
+} from '@mui/x-data-grid-pro/internals';
+import { GridAggregationRowOverlay } from './GridAggregationRowOverlay';
+
+const useUtilityClasses = () => {
+  const slots = {
+    root: ['pinnedRows'],
+  };
+  return composeClasses(slots, getDataGridUtilityClass, {});
+};
+
+export function GridPinnedRows({ position, virtualScroller }: GridPinnedRowsProps) {
+  const classes = useUtilityClasses();
+  const apiRef = useGridPrivateApiContext();
+
+  const renderContext = useGridSelector(apiRef, gridRenderContextSelector);
+  const pinnedRowsData = useGridSelector(apiRef, gridPinnedRowsSelector);
+  const rows = pinnedRowsData[position];
+
+  const pinnedRenderContext = React.useMemo(
+    () => ({
+      firstRowIndex: 0,
+      lastRowIndex: rows.length,
+      firstColumnIndex: renderContext.firstColumnIndex,
+      lastColumnIndex: renderContext.lastColumnIndex,
+    }),
+    [rows, renderContext.firstColumnIndex, renderContext.lastColumnIndex],
+  );
+
+  if (rows.length === 0) {
+    return null;
+  }
+
+  const pinnedRows = virtualScroller.getRows({
+    position,
+    rows,
+    renderContext: pinnedRenderContext,
+  });
+
+  return (
+    <div className={clsx(classes.root, gridClasses[`pinnedRows--${position}`])} role="presentation">
+      {pinnedRows}
+      {position === 'bottom' && <GridAggregationRowOverlay />}
```

Comment:
There's a missing border with pinned row cells when also in a pinned column:
<img width="836" alt="Screenshot 2025-01-30 at 10 36 21" src="https://github.com/user-attachments/assets/83f879ff-aad5-4e46-bce0-c85fcca05a81" />

Should look like:
<img width="832" alt="Screenshot 2025-01-30 at 10 41 22" src="https://github.com/user-attachments/assets/8ba9b103-998c-49b5-ae18-7492551aafd3" />